### PR TITLE
Chore: Updated flavor definition to allow resource separation

### PR DESCRIPTION
**Pull request by aanorbel**
## Proposed Changes

  - Define `flavorDimensions` for `provider` where specific resources for `ooni` or `dw` can be placed.
  - Provide base strings in the main resources `English` where more languages can be added in the provider folders and supported languages added.
  - Move `ooni` specific translations to the right folder.

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -7,8 +7,8 @@ jobs:
     strategy:
       matrix:
         version:
-            - "StableFullRelease"
-            - "StableFdroidRelease"
+            - "OoniStableFullRelease"
+            - "OoniStableFdroidRelease"
```
Comment (bassosimone) [inline .github/workflows/build.yml:11]: TODO(@aanorbel): please, update the main `README.md` given that it contains build instructions and lists all the variants as an introduction to the reader.

TODO(@bassosimone): make sure we also update the monorepo scripts once we merge.

File: app/build.gradle [hunk 2]
```diff
@@ -79,16 +89,10 @@ android {
...
-            resValue "string", "APP_NAME", "News Media Scan"
-            buildConfigField 'String', 'BASE_SOFTWARE_NAME', '"news-media-scan-android"'
-            buildConfigField "boolean", "SHOW_DISABLED_CARDS", "false"
-            //versionName '0.0.1'
-            //versionCode 1
-        }
+    }
+    applicationVariants.all { variant ->
+        variant.resValue "string", "APP_ID", "\"${applicationId}\""
+        // TODO(aanorbel): Use similar solution to set `APP_NAME` in the format `provider` `testing`.
```
Comment (bassosimone) [inline app/build.gradle:95]: Do you want to do this as part of the same PR or as part of another PR? (I am flexible)

File: app/src/main/java/org/openobservatory/ooniprobe/common/PreferenceManager.java [hunk 3]
```diff
@@ -328,9 +329,12 @@ public boolean isAutomaticTestEnabled() {
 	public String getSettingsLanguage() {
 		String language = sp.getString(r.getString(R.string.language_setting), Locale.getDefault().getLanguage());
 		if (language.equals("auto")) {
-			return Locale.getDefault().getLanguage();
+			language = Locale.getDefault().getLanguage();
 		}
-		return language;
+		if (new ArrayList<>(Arrays.asList(r.getStringArray(R.array.language_sort_options_values))).contains(language)){
```
Comment (bassosimone) [inline app/src/main/java/org/openobservatory/ooniprobe/common/PreferenceManager.java:334]:
```suggestion
		if (new ArrayList<>(Arrays.asList(r.getStringArray(R.array.language_sort_options_values))).contains(language)) {
```

File: app/src/ooni/dev/readme.md [hunk 4]
```diff
@@ -0,0 +1,3 @@
+# OONI Dev
+This is a collection of resources to be used by the OONI specific flavors of the application.
+A corresponding directory exits in [app/src/dw](../../dw).
```
Comment (bassosimone) [inline app/src/ooni/dev/readme.md:3]: Would you mind calling the `readme.md` files `README.md`?

File: app/src/dw/dev/google-services.json [hunk 5]
```diff
@@ -0,0 +1,56 @@
+{
```
Comment (bassosimone) [inline app/src/dw/dev/google-services.json:1]: Please, add a `README.md` also to this directory, for consistency.

**Review by bassosimone — CHANGES_REQUESTED** [review]
Looks good to me. I documented minor improvements and TODOs for me and you.

File: app/build.gradle [hunk 6]
```diff
@@ -47,8 +47,20 @@ android {
         }
     }
 
-    flavorDimensions 'testing', 'license'
+    flavorDimensions = ['provider', 'testing', 'license']
```
Comment (hellais) [inline app/build.gradle:50]: brand?

**Review by hellais — COMMENTED** [review]
This looks great!
Small nitpick comment on the flavorDimension value, but otherwise LGTM 🐳

File: app/build.gradle (hunk 2 above)
Comment (aanorbel) [reply app/build.gradle:95]: This PR

**Pull request merged**